### PR TITLE
Feature/mfkans

### PR DESCRIPTION
**Pull request by brunopjacob**
Adding example of multi-fidelity KANs

### CODE REVIEW

**Review by RBirmiwal — APPROVED** [review]

**Comment by brunopjacob** [comment]
@drgona  @RBirmiwal please take a look and let me know if this looks good! If yes, we are good to merge. Thank you!

File: src/neuromancer/trainer.py [hunk 1]
```diff
@@ -250,6 +252,11 @@ def train(self):
                             alpha_loss = node.callable.get_alpha_loss()
                             output[self.train_metric] += alpha_loss
 
+                    for node in self.model.nodes:
```
Comment (drgona) [inline src/neuromancer/trainer.py:255]: What is the motivation to modify the trainer itself to include this KAN regularization loss? 
This does not seem very modular solution. We can simply instantiate the regularization loss as a separate loss term during problem formulation rather than handling it internally via trainer.

**Review by drgona — CHANGES_REQUESTED** [review]

**Comment by drgona** [comment]
@brunopjacob are we still planning to finish the mfkans?
